x-pack/legacy/plugins/reporting/export_types/printable_pdf/server/execute_job/compatibility_shim.js

```diff
@@ -4,41 +4,10 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import url from 'url';
-import { getAbsoluteUrlFactory } from '../../../../common/get_absolute_url';
-import { i18n } from '@kbn/i18n';
-
-export function compatibilityShimFactory(server) {
-  const getAbsoluteUrl = getAbsoluteUrlFactory(server);
-
-  const getSavedObjectAbsoluteUrl = (job, savedObject) => {
-    if (savedObject.urlHash) {
-      return getAbsoluteUrl({ hash: savedObject.urlHash });
-    }
-
-    if (savedObject.relativeUrl) {
-      const { pathname: path, hash, search } = url.parse(savedObject.relativeUrl);
-      return getAbsoluteUrl({ basePath: job.basePath, path, hash, search });
-    }
-
-    if (savedObject.url.startsWith(getAbsoluteUrl())) {
-      return savedObject.url;
-    }
-
-    throw new Error(i18n.translate('xpack.reporting.exportTypes.printablePdf.compShim.unableToGenerateReportErrorMessage', {
-      defaultMessage: `Unable to generate report for url {savedObjUrl}, it's not a Kibana URL`,
```

Comment:
Absolute URLs are derived in execute_job, so I'm not sure how this error would happen now. There's no new analogy for this error in the changes in this PR.